onnxruntime/test/providers/provider_test_utils.cc

```diff
@@ -334,10 +357,12 @@ void Check(const OpTester::Data& expected_data, const Tensor& output_tensor,
                   output_tensor.Shape().ToString() + "] for " +
                   expected_data.def_.Name());
 
-  CheckDispatch<bool, float, double, uint8_t, uint16_t, uint32_t, uint64_t,
-                int8_t, int16_t, int32_t, int64_t, std::string, MLFloat16,
-                BFloat16>(output_tensor.DataType(), expected_data,
-                          output_tensor, provider_type);
+  utils::MLTypeCallDispatcher<bool, float, double, uint8_t, uint16_t, uint32_t, uint64_t,
+                              int8_t, int16_t, int32_t, int64_t, std::string, MLFloat16,
+                              BFloat16>
+      t_disp(output_tensor.GetElementType());
+
+  t_disp.Invoke<TensorCheck>(expected_data, output_tensor, provider_type);
```

Comment:
how about
`t_disp.Invoke<TensorCheck>(expected_data.data_.Get<Tensor>(), output_tensor, provider_type, make_params(expected_data))`
so we only need to provide a single operator()() in the TensorCheck structs?